### PR TITLE
JENKINS-31164 - Fixed issueTypeId of Bug , so this plugin could work with latest version of JIRA

### PR DESCRIPTION
**Pull request by sam-shen**
fixed issueTypeId of Bug ,so this plugin could work with latest version of JIRA

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/jira-plugin/109)

### CODE REVIEW

**Comment by warden** [comment]
This would break previous versions of JIRA which have different BUG ID.
This cannot be a constant value but a value that is populated via REST API from `rest/api/2/issuetype`

**Comment by sam-shen** [comment]
Add getIssueTypeId(String issueTypeName), should work with previous versions of JIRA which have different BUG ID with same "Bug" name.

**Comment by johnou** [comment]
I believe my pull request supersedes this one https://github.com/jenkinsci/jira-plugin/pull/112 👍

File: src/main/java/hudson/plugins/jira/JiraRestService.java [hunk 1]
```diff
@@ -222,11 +223,27 @@ public void releaseVersion(String projectKey, Version version) {
         }
     }
 
+    public Long getIssueTypeId(String issueTypeName) {
+    	List<IssueType> issueTypeList = new ArrayList<IssueType>();
+        Long issueTypeId = 10004L;
+        
+    	issueTypeList = this.getIssueTypes();
+    	for(IssueType issue : issueTypeList) {
+    		LOGGER.warning("issueType: " + issue.getName() + "  "+ issue.getId());
```
Comment (johnou) [inline src/main/java/hudson/plugins/jira/JiraRestService.java]: This logging will cause pretty bad spam..

**Comment by johnou** [comment]
Closing in favour of https://github.com/jenkinsci/jira-plugin/pull/112

**Pull request closed without merging**